File: study/src/api/consultation.ts
import request from '@/utils/request'

export interface ConsultationCreateForm {
  doctorId: number
  consultationType: 'text' | 'video'
  symptom: string
  description: string
  images?: string
  medicalHistory?: string
  currentMedication?: string
}

export interface ConsultationQueryParams {
  status?: string
}

export interface DoctorVO {
  id: number
  name: string
  title: string
  avatar: string
  departmentName?: string
}

export interface ConsultationVO {
  id: number
  consultationNo: string
  doctor: DoctorVO
  consultationType: string
  symptom: string
  description: string
  images?: string
  medicalHistory?: string
  currentMedication?: string
  status: string
  statusText: string
  consultationFee?: number
  patientName?: string
  diagnosis?: string
  doctorReply?: string
  prescriptionSuggestion?: string
  createdAt: string
}

export interface ApiResponse<T> {
  code: number
  msg: string
  data: T
}

export const createConsultation = (data: ConsultationCreateForm) => {
  return request.post<ApiResponse<ConsultationVO>>('/api/consultation/create', data)
}

export const getConsultations = (params?: ConsultationQueryParams & { page?: number; size?: number }) => {
  const queryParams: Record<string, any> = {}
  if (params?.status) {
    queryParams.status = params.status
  }
  if (params?.page) {
    queryParams.page = params.page
  }
  if (params?.size) {
    queryParams.size = params.size
  }
  return request.get<ApiResponse<any>>('/api/consultation/list', { params: queryParams })
}

export const getConsultationById = (id: number) => {
  return request.get<ApiResponse<ConsultationVO>>(`/api/consultation/${id}`)
}

export const cancelConsultation = (id: number) => {
  return request.put<ApiResponse<ConsultationVO>>(`/api/consultation/${id}/cancel`)
}

export const updateConsultation = (id: number, data: Partial<ConsultationVO>) => {
  return request.put<ApiResponse<ConsultationVO>>(`/api/consultation/${id}`, data)
}

File: demo/src/main/java/com/example/usergenerator/entity/Consultation.java
package com.example.usergenerator.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@TableName("consultation")
public class Consultation {

    @TableId(value = "id", type = IdType.AUTO)
    private Long id;

    @TableField("consultation_no")
    private String consultationNo;

    @TableField("user_id")
    private Long userId;

    @TableField("doctor_id")
    private Long doctorId;

    @TableField("consultation_type")
    private String consultationType;

    @TableField("symptom")
    private String symptom;

    @TableField("description")
    private String description;

    @TableField("images")
    private String images;

    @TableField("medical_history")
    private String medicalHistory;

    @TableField("current_medication")
    private String currentMedication;

    @TableField("doctor_reply")
    private String doctorReply;

    @TableField("patient_message")
    private String patientMessage;

    @TableField("diagnosis")
    private String diagnosis;

    @TableField("status")
    private String status;

    @TableField("created_at")
    private LocalDateTime createdAt;

    @TableField("updated_at")
    private LocalDateTime updatedAt;

    @TableField("deleted")
    private Integer deleted;
}